### PR TITLE
Add tiatoolbox tutorial

### LINKED ISSUE #2669: 💡 [REQUEST] - A Tutorial on Whole Slide Image Classification using PyTorch and TIAToolbox
### 🚀 Descirbe the improvement or the new tutorial

Whole Slide Images are the digital data format from which pathologists and computational pathology researchers investigate cancer growth. To due their enormous image resolutions and and file size (in the order of several gigabytes), conventional image processing methods do not work effectively. This is why we propose writing this tutorial: to (a) explain how to load WSIs using TIAToolbox, which helps process such slides with speed and efficiency using its pyramid stack structure, and (b) show how you can use `torchvision` models can to analyse WSIs. We believe this tutorial will be useful to the PyTorch community, especially who is interested in using PyTorch models tackle cancer tissue research.

### Existing tutorials on this topic

The tutorial will be adapted from our [WSI classification example](https://tia-toolbox.readthedocs.io/en/latest/_notebooks/jnb/05-patch-prediction.html). 

### Additional context

**Category: Image and Video**

Written by Tissue Image Analytics Centre (TIA) and invited by @carljparker as part of the PyTorch Docathon H2 2023.
cc @datumbox @nairbv @fmassa @NicolasHug @YosuaMichael @sekyondaMeta @svekars @carljparker @kit1980 @subramen @measty @behnazelhaminia @DavidBAEpstein @shaneahmed @msaroufim

### PR DESCRIPTION
**Pull request by measty**
Fixes #2669

## Description
Adds a tutorial for using TIAToolbox, a python/pytorch library for working with whole slide images (large medical images).

## Checklist

- [x ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x ] Only one issue is addressed in this pull request
- [x ] Labels from the issue that this PR is fixing are added to this pull request
- [ x] No unnecessary issues are included into this pull request.

cc @datumbox @nairbv @fmassa @NicolasHug @YosuaMichael @sekyondaMeta @svekars @carljparker @kit1980 @subramen

### CODE REVIEW

**Comment by carljparker** [comment]
Add @NicolasHug as reviewer, in addition to @subramen

File: intermediate_source/tiatoolbox_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,994 @@
+Whole Slide Image Classification Using PyTorch and TIAToolbox
+=============================================================
+
+.. tip::
+   To get the most of this tutorial, we suggest using this
+   `Colab Version <https://colab.research.google.com/drive/1qkTBbMrECLjVAWwg9SJHKFgmSUyv3djE?usp=sharing>`_. This will allow you to experiment with the information presented below.
```
Comment (malfet) [inline intermediate_source/tiatoolbox_tutorial.rst:6]: Why can't we turn this into `.py` file and generate tutorial on the fly?

Comment (measty) [reply intermediate_source/tiatoolbox_tutorial.rst:6]: Hi malfet,
We initially had it set up like this, but encountered some issues with dependency conflicts between tiatoolbox/histoencoder and the requirements of the pytorch tutorials repo. For example, tiatoolbox uses sphinx for documentation also, and there were some conflicts arising from that. And histoencoder, which is one of only a few available pathology-image-specific 'foundation models' which we use to demonstate feature extraction on whole slide images using tiatoolbox, requires a specific version of timm. Rather than trying to resolve these conflicts and potentially having the conflicts re-appearing whenever the tutorials requirements evolve, we thought it better to submit as a .rst tutorial.

**Review by subramen — APPROVED** [review]

**Review by malfet — CHANGES_REQUESTED** [review]
If one can not autogenerate .ipynb from the .py file, please check it into the tutorials repo and reference it rather than https://colab.research.google.com/drive/1qkTBbMrECLjVAWwg9SJHKFgmSUyv3djE?usp=sharing (because if/when author decided to stop sharing it it will not be accessible to the reader)

File: intermediate_source/tiatoolbox_tutorial.rst (hunk 1 above)
Comment (measty) [reply intermediate_source/tiatoolbox_tutorial.rst:6]: We were invited to contribute a tutorial on the use of TIAToolbox, as an example of pytorch at work in a specific application domain. TIAToolbox is a toolbox for computational pathology, so by it's nature it will be of most interest to members of that community. It is one that is growing very rapidly as we reach a critical point of available scanned tissue images, as most hospitals are moving to digital workflows.

The toolbox is quite flexible and the feature extraction, for example, could be done with any pytorch model - we could have demonstrated generating a feature map with a generic ImageNet-pretrained backbone and it would give reasonable results, but given its most likely audience is people interested in digital pathology, we thought it would be especially valuable to demonstrate how to use a pathology-specific backbone within the whole slide image processing pipelines that tiatoolbox provides.

**Comment by measty** [comment]
> If one can not autogenerate .ipynb from the .py file, please check it into the tutorials repo and reference it rather than https://colab.research.google.com/drive/1qkTBbMrECLjVAWwg9SJHKFgmSUyv3djE?usp=sharing (because if/when author decided to stop sharing it it will not be accessible to the reader)

Should this be done by making a PR against the `gh-pages` branch, or in some other way?

**Comment by malfet** [comment]
> 
> Should this be done by making a PR against the `gh-pages` branch, or in some other way?

No, just add one to this PR in a folder where it will be published

**Comment by carljparker** [comment]
@malfet: Are we good to merge (provided CI is clean)? @measty added the notebook to an `_static` subdirectory.

**Comment by measty** [comment]
> @malfet: Are we good to merge (provided CI is clean)? @measty added the notebook to an `_static` subdirectory.

Sorry, I noticed I forgot to change the link to point to the new location when I added the notebook to this PR. Should be all set now from my end.
